jvcl/run/JvAppIniStorage.pas

```diff
@@ -704,8 +706,21 @@ procedure TJvCustomAppIniStorage.DeleteSubTreeInt(const Path: string);
           IniFile.EraseSection(Sections[I])
       else
         for I := 0 to Sections.Count - 1 do
-          if Pos(TopSection, Sections[I]) = 1 then
-            IniFile.EraseSection(Sections[I]);
+        begin
+          // --- start bugfix ---
+          // What if the section name is a prefix to another unrelated section?
+          // [bla/blub]
+          // [bla/blubber]
+          // I now delete [bla/blub] which would also delete [bla/blubber] which is a bug.
+          // -- 2019-03-26 twm
```

Comment:
Same here, and the name of the committer is even useless, it shows up in the commit log. As it is, I'm unable to merge this pull request.